lib/puppet/provider/rabbitmq_binding/rabbitmqadmin.rb

```diff
@@ -51,11 +51,13 @@ def self.instances
         end
         if (source_name != '')
           binding = {
+            :source           => source_name,
+            :dest             => destination_name,
+            :vhost            => vhost,
             :destination_type => destination_type,
             :routing_key      => routing_key,
             :arguments        => JSON.parse(arguments),
             :ensure           => :present,
-            :name             => "%s@%s@%s" % [source_name, destination_name, vhost],
           }
           resources << new(binding) if binding[:name]
         end
```

Comment:
Prefetch should match on all 4 properties, not on `.name` like

`if provider = bindings.find { |route| route.source == source && route.dest == dest && ... }`